app/controllers/registered_applications_controller.rb

```diff
@@ -0,0 +1,29 @@
+class RegisteredApplicationsController < ApplicationController
+
+  def index
+    @registered_applications = RegisteredApplication.all
+  end
+
+  def new
+    @registered_application = RegisteredApplication.new
+  end
+
+  def create
+    @registered_application = RegisteredApplication.new
+    @registered_application.name = params[:registered_application][:name]
+    @registered_application.url = params[:registered_application][:url]
+
+    if @registered_application.save
+      redirect_to registered_applications_path
+    else
+      render :new
+    end
+  end
+
+  def destroy
+    @registered_application = RegisteredApplication.find(params[:id])
+    @registered_application.destroy!
+    redirect_to registered_applications_path
+  end
```

Comment:
With some creative use of the browser dev tools to modify the `params[:id]` value sent to the `destroy` action, its possible for a logged-in user to delete other users` registered applications (we can try this on a call if you're interested).

To protect against this vulnerability, limit the `.find(params[:id])` call to only search the `current_user` registered applications:

```diff
-@registered_application = RegisteredApplication.find(params[:id])
+@registered_application = current_user.registered_applications.find(params[:id])
```

The SQL query generated by `current_user.registered_applications.find(params[:id])` will include a `WHERE registered_applications.user_id = current_user_id` clause to limit the query results to the current user's registered applications only.